packages/core/core/src/Parcel.js

```diff
@@ -25,20 +29,39 @@ type ParcelOpts = {
   env?: {[string]: ?string}
 };
 
-export default class Parcel {
+export default class Parcel extends EventEmitter {
   options: ParcelOpts;
   entries: Array<string>;
   rootDir: string;
   assetGraphBuilder: AssetGraphBuilder;
   bundlerRunner: BundlerRunner;
   farm: WorkerFarm;
+  server: Server;
+  pending: boolean;
+  error: Error | null;
   runPackage: (bundle: Bundle) => Promise<any>;
 
   constructor(options: ParcelOpts) {
+    super();
+
     let {entries} = options;
     this.options = options;
     this.entries = Array.isArray(entries) ? entries : [entries];
     this.rootDir = getRootDir(this.entries);
+
+    this.normaliseCliOptions();
+  }
+
+  normaliseCliOptions() {
+    this.options.cliOpts.cacheDir =
+      this.options.cliOpts.cacheDir || DEFAULT_CACHE_DIR;
```

Comment:
I kinda liked doing this normalization in the individual places rather than all in one place... What do you think? Why did you move them here?